src/main/java/uk/gov/hmcts/reform/professionalapi/controller/external/ProfessionalExternalUserController.java

```diff
@@ -78,14 +82,71 @@
+        ServiceAndUserDetails serviceAndUserDetails = (ServiceAndUserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        boolean isRolePuiUserManager = organisationIdentifierValidatorImpl.ifUserRoleExists(serviceAndUserDetails.getAuthorities(), "pui-user-manager");
+        ResponseEntity<?> profUsersEntityResponse;
+
+        if (!isRolePuiUserManager) {
+            if (StringUtils.isEmpty(status)) {
+                status = "Active";
+            }
+            profExtUsrReqValidator.validateStatusIsActive(status);
+        }
+
+        profUsersEntityResponse = searchUsersByOrganisation(organisationIdentifier, showDeleted, true, status, page, size);
+
+        return profUsersEntityResponse;
+    }
+
+    @ApiOperation(
+            value = "Retrieves the user with the given email address if organisation is active or showDeleted flag ",
+            authorizations = {
+                    @Authorization(value = "ServiceAuthorization"),
+                    @Authorization(value = "Authorization")
+            }
+    )
+    @ApiParam(
+            name = "email",
+            type = "string",
+            value = "The email of the desired user to be retrieved",
+            required = false
+    )
+    @ApiResponses({
+            @ApiResponse(
+                    code = 200,
+                    message = "A professional user along with their details",
+                    response = ProfessionalUsersEntityResponse.class
+            ),
+            @ApiResponse(
+                    code = 400,
+                    message = "An invalid email was provided"
+            ),
+            @ApiResponse(
+                    code = 403,
+                    message = "Invalid authorization"
+            ),
+            @ApiResponse(
+                    code = 404,
+                    message = "No user was found with the provided email address"
+            )
+    })
+    @GetMapping(
+            value = "/user",
+            produces = MediaType.APPLICATION_JSON_UTF8_VALUE
+    )
+    @Secured({"pui-user-manager"})
+    public ResponseEntity<ProfessionalUsersResponse> findUserByEmail(@ApiParam(hidden = true) @OrgId String organisationIdentifier,
+                                                                     @ApiParam(name = "email", required = false) @RequestParam(value = "email", required = false) String email) {
+
+        ResponseEntity<ProfessionalUsersResponse> profUsersEntityResponse = null;
+        log.info("ProfessionalExternalUserController::findUserByEmail:" + email);
+        isValidEmail(email);
         ServiceAndUserDetails serviceAndUserDetails = (ServiceAndUserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
         boolean isRolePuiUserManager = organisationIdentifierValidatorImpl.ifUserRoleExists(serviceAndUserDetails.getAuthorities(), "pui-user-manager");
 
```

Comment:
You are logging the email further down. Please remove necessary logging such as this, as it will spam the logs and slow down performance.

Reply:
This will be removed